azurelinuxagent/ga/exthandlers.py

```diff
@@ -243,6 +243,20 @@ class ExtensionRequestedState(object):
     Uninstall = u"uninstall"
 
 
+class GoalStateState(object):
+    """
+    This is an Enum to define the State of the GoalState as a whole. This is reported as part of the
+    'vmArtifactsAggregateStatus.goalStateAggregateStatus' in the status blob.
+    Note: not to be confused with the State of the ExtHandler which reported as part of 'handlerAggregateStatus'
+    """
+    Success = "Success"
+    Failed = "Failed"
+    Initialize = "Initialize"
```

Comment:
Its basically used when we initialize a new goalstate. Its primarily for use in a multi-threaded environment and doesnt make a lot of sense in a single-threaded case because we wont report status until the goalstate processing is completed

Reply:
Let's sync offline. Not sure I understand the difference between Initialize and Transitioning